cloudferrylib/os/actions/check_cloud.py

```diff
@@ -12,153 +12,141 @@
+                     subnet_info['subnet']['name'])
             nt_client.neutron_client.delete_subnet(subnet_id)
 
     @contextlib.contextmanager
     def create_image(self, gl_client, image_info):
-        LOG.info("Creating %s image.", image_info['name'])
+        LOG.info("Creating image '%s'...", image_info['name'])
         image = gl_client.create_image(**image_info)
         try:
-            yield image
+            yield image.id
         finally:
-            LOG.info("Deleting %s image.", image_info['name'])
+            LOG.info("Deleting previously created image '%s'...", image.name)
             gl_client.delete_image(image.id)
 
     @contextlib.contextmanager
     def create_volume(self, cn_client, volume_info):
-        LOG.info("Creating %s volume.", volume_info['display_name'])
-        tenant = volume_info.pop('tenant')
-        client = cn_client.get_client(tenant=tenant)
-        volume = client.volumes.create(**volume_info)
+        LOG.info("Creating volume '%s'...", volume_info['display_name'])
+        volume = cn_client.create_volume(**volume_info)
+        cn_client.wait_for_status(volume.id, cn_client.get_status, 'available')
         try:
             yield
         finally:
-            LOG.info("Deleting %s volume.", volume_info['display_name'])
-            self.wait_for_volume_to_become_available(cn_client, volume.id)
+            LOG.info("Deleting previously created volume '%s'...",
+                     volume.display_name)
             cn_client.delete_volume(volume.id)
 
     @contextlib.contextmanager
     def create_instance(self, nv_client, instance_info):
-        LOG.info("Creating %s instance.", instance_info['name'])
-        instance_id = nv_client.nova_client.servers.create(**instance_info)
-        nv_client.wait_for_status(instance_id, nv_client.get_status, 'active',
+        LOG.info("Creating instance '%s'...", instance_info['name'])
+        instance = nv_client.nova_client.servers.create(**instance_info)
+        nv_client.wait_for_status(instance.id, nv_client.get_status, 'active',
                                   timeout=300)
         try:
             yield
         finally:
-            LOG.info("Deleting instance.")
-            nv_client.nova_client.servers.delete(instance_id)
+            LOG.info("Deleting previously created instance '%s'...",
+                     instance.name)
+            nv_client.nova_client.servers.delete(instance.id)
             self.wait_for_instance_to_be_deleted(nv_client=nv_client,
-                                                 instance_id=instance_id,
+                                                 instance=instance,
                                                  timeout=300)
 
-    @classmethod
-    def wait_for_instance_to_be_deleted(cls, nv_client, instance_id,
-                                        timeout=60):
+    @staticmethod
+    def wait_for_instance_to_be_deleted(nv_client, instance, timeout=60):
```

Comment:
Since we're rewriting it, I'd suggest using `cloudferrylib.utils.retrying` for retries